lib/automatic/api_client/client_delegator.rb

```diff
@@ -0,0 +1,21 @@
+module Automatic
+  class ApiClient
+    class ClientDelegator
+      attr_reader :api_client, :delegated_class
+
+      def initialize(api_client, delegated_class)
+        @api_client = api_client
+        @delegated_class = delegated_class
+      end
+
+      def method_missing(method_sym, *arguments, &block)
+        super unless delegated_class.respond_to?(method_sym)
+
+        arguments[1 + delegated_class.method(method_sym).arity] ||= {}
```

Comment:
My initial thought was to change this gem to move to an instance-based approach, and gut the entire class-level stuff. You could also create some kind of singleton class that handles the existing case of "only one client, and it's at the class level" to handle the legacy stuff. If that makes sense.